kompari-cli/src/main.rs

```diff
@@ -89,16 +93,30 @@ fn main() -> kompari::Result<()> {
     report_config.set_left_title(args.left_title);
     report_config.set_right_title(args.right_title);
 
-    match args.command {
-        Command::Report(args) => {
+    (diff_config, report_config)
+}
+
+fn main() -> kompari::Result<()> {
+    let args = Args::parse();
+
+    match args {
+        Args::Report(args) => {
+            let (diff_config, mut report_config) = make_diff_config(args.diff_args);
             let diff = diff_config.create_diff()?;
-            report_config.set_embed_images(args.embed_images);
+            report_config.set_embed_images(args.args.embed_images);
             let report = render_html_report(&report_config, diff.results())?;
-            let output = args.output;
+            let output = args.args.output.unwrap_or("report.html".into());
             std::fs::write(&output, report)?;
             println!("Report written into '{}'", output.display());
         }
-        Command::Review(args) => start_review_server(&diff_config, &report_config, args.port)?,
+        Args::Review(args) => {
+            let (diff_config, report_config) = make_diff_config(args.diff_args);
+            start_review_server(&diff_config, &report_config, args.args.port)?
+        }
+        Args::SizeCheck(args) => {
+            let results = check_size_optimizations(&args.path, args.args.optimize)?;
```

Comment:
It would be good to also print how long optimising took (probably just as a total time)